test/SlowTests/Issues/RavenDB-17157.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Session;
+using Raven.Client.Json;
+using SlowTests.Core.Utils.Entities;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17157 : ClusterTestBase
+    {
+        public RavenDB_17157(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void OnSessionCreated_WaitForIndexesAfterSaveChanges_Test()
+        {
+            using (var store = GetDocumentStore())
+            {
+                store.OnSessionCreated += (sender, sessionCreatedEventArgs) =>
+                {
+                    sessionCreatedEventArgs.Session.WaitForIndexesAfterSaveChanges();
+                };
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User { Name = "Foo" });
+                    session.Store(new User { Name = "Bar" });
+                    session.SaveChanges();
+
+                    QueryStatistics stats;
+                    var results = session.Query<User>()
```

Comment:
auto indexes on creation will wait up to 15s to get the results, so this tests does not do what you think. Better option would be to:
1. Create static index
2. Stop the indexing
3. Do SaveChanges and wait for exception (use reasonable timeout, like 500ms)

Reply:
i fixed this test and added a test with disabled index.